### PR TITLE
Fix Site Admin Repositories page from over-querying and hitting rate limits

### PR DESCRIPTION
**Pull request by varsanojidan**
Fixes https://github.com/sourcegraph/sourcegraph/issues/55236

The main issue was that there was a PollInterval on the SettingsAreaRepositories query so it kept spamming it per repo and eventually it could get you rate limited.

That entire query was to refetch the repo when someone hit tge reclone button, now it just uses the refetch callback from the parent when the re-clone button is hit, much much much less expensive.

## Test plan

manual.

### CODE REVIEW

File: client/web/src/site-admin/RepositoryNode.tsx [hunk 1]
```diff
@@ -109,17 +99,10 @@ export const RepositoryNode: React.FunctionComponent<React.PropsWithChildren<Rep
...
-            variables: { name: node.name },
-            pollInterval: 3000,
-        }
-    )
+
     const recloneAndFetch = async (): Promise<void> => {
         await recloneRepository()
-        await refetch()
-        updateNodeFromData(node, data)
+        await refetchAllRepos()
```
Comment (mrnugget) [inline client/web/src/site-admin/RepositoryNode.tsx:105]: Why do we have to refetch all repositories on the page if we just recloned a single repository?

File: client/web/src/site-admin/RepositoryNode.tsx [hunk 2]
```diff
@@ -109,17 +99,10 @@ export const RepositoryNode: React.FunctionComponent<React.PropsWithChildren<Rep
         UPDATE_MIRROR_REPOSITORY,
         { variables: { repository: node.id } }
     )
-    const { data, refetch } = useQuery<SettingsAreaRepositoryResult, SettingsAreaRepositoryVariables>(
```
Comment (mrnugget) [inline client/web/src/site-admin/RepositoryNode.tsx:112]: If the problem is that this query is always sent, for each repository, and then continues polling -- why not try to find out how to run the query only when `recloneAndFetch` is called and remove polling? There is `useLazyQuery`, for example.

File: client/web/src/site-admin/RepositoryNode.tsx [hunk 3]
```diff
@@ -83,20 +80,13 @@ const repoCloned = (repo: SiteAdminRepositoryFields): boolean =>
 
 interface RepositoryNodeProps {
     node: SiteAdminRepositoryFields
+    refetchAllRepos: any
```
Comment (sashaostrikov) [inline client/web/src/site-admin/RepositoryNode.tsx:83]: Hmm, can we get a better type here, maybe?

**Review by sashaostrikov — COMMENTED** [review]
The problem described in the issue is that we're polling 20 queries instead of 1.

One part of this PR fixes that, which is good. I don't think we should remove polling altogether because the page may become "stale" as not only the `recloneAndFetch` is responsible for changing the repo cloning status.

The second part is changing the "per-repo" query to the "batch" query when a single repo is recloned. I'd revert this change for the sake of having a finer-grained query.

**Comment by mrnugget** [comment]
> I don't think we should remove polling altogether because the page may become "stale" as not only the `recloneAndFetch` is responsible for changing the repo cloning status.

We do have separate polling on that page though, for all repositories:

https://github.com/sourcegraph/sourcegraph/blob/f21bd8c6ef335be9f912e6b1b6132965f34f2565/client/web/src/site-admin/SiteAdminRepositoriesContainer.tsx#L280

Do you mean in addition to that you want to poll the single repo node?

**Review by eseliger — APPROVED** [review]

**Pull request merged**